util/topgen/drg3.py

```diff
@@ -0,0 +1,233 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+"""Generating random variables by using a cryptographically strong PRNG.
+
+A class for generating random numbers and permutations.
+This class is based on a cryptographically secure PRNG CTR_DRBG described in
+NIST SP 800-90A, section 10.2.1
+https://csrc.nist.gov/publications/detail/sp/800-90a/rev-1/final
+This class guarentees forward and backward secrecy (PRNG output cannot be
+computed from any previous or future outpus) as well as enhanced backward
+secrecy (previous PRNG outputs cannot be computed from a compromised
+internal state).
+"""
+
+import logging as log
+import sys
+from Crypto.Cipher import AES
+from math import ceil as _ceil
+from math import log as _log
+
+
+class drg3():
+
+    blocklen = 128
+    keylen = 128
+    seedlen = 256  # blocklen + keylen
+    ctr_len = 128  # blocklen
+    security_strength = 128
+    min_length = 128  # security_strength
+    outlen = 128 // 8
+
+    # Maximal number of requests before reseed is required
+    RESEED_MAX = 2 ** 48
+
+    def __init__(self):
+        """Initialize an instance.
+
+        Default state value is 0.
+        """
+
+        # A 128 bit PRNG output
+        self.returned_bits = []
+
+        # CTR_DRBG Internal state as defined in NIST SP 800-90A, page 50
+        # The state consists of:
+        #     - The value V of 128 bits
+        #     - The value Key of 128 bits
+        #     - Reseed counter
+        # Administrative information such as security_strength and
+        # prediction_resistance_flag are missing in this implementation
+        # because they are not used by the consuming application.
+        self.V = 0
+        self.Key = 0
+        self.reseed_counter = 1
```

Comment:
This should actually be done the instantiate function. Is it possible to initialize these values to something like `null` here?

Reply:
Good point. I've changed all integer values to None and replaced update function with `drg3.reseed(0)`.